lib/x11utils.pm

```diff
@@ -197,7 +197,13 @@ sub handle_login {
 
     save_screenshot();
     # wait for DM, avoid screensaver and try to login
-    send_key_until_needlematch('displaymanager', 'esc', 30, 3);
+    if (check_var('DESKTOP', 'kde')) {
+        # Pressing escape makes the text field lose focus
+        send_key_until_needlematch('displaymanager', 'shift', 30, 3);
+    }
+    else {
+        send_key_until_needlematch('displaymanager', 'esc', 30, 3);
```

Comment:
did you keep this "esc" because it's still required or because you did not test? If the latter, well, please test :) If the first, I recommend to write as:

```perl
# Pressing escape makes the text field lose focus in sddm
send_key_until_needlematch('displaymanager', check_var('DESKTOP', 'kde') ? 'shift' : 'esc', 30, 3);
```

Reply:
I kept the esc because I'd have to test everything which does a login - that's a lot.

I have no idea whether GNOME over vnc on s390x also accepts shift and so on...

Reply:
Yeah but that's messy. I suggest that either you test that which you can simply by triggering tests against your PR with http://open.qa/docs/#_triggering_tests_based_on_an_any_remote_git_refspec_or_open_github_pull_request or wait for helpful QA engineers to trigger according tests against your PR or you only test basics like "gnome on Tumbleweed x86_64" and we merge and hope that any failures elsewhere are handled by reviewers that find the "Investigation" tab to see your change easily and tell you :)